### PR TITLE
Single post coments

### PR DESCRIPTION
**Pull request by vkravchuk2**

### CODE REVIEW

**Review by attiaahmedtarek — APPROVED** [review]

File: src/components/Coments/Coment.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Avatar } from "@mui/material"
+import './Coment.css'
+import { TextField } from '@material-ui/core'
+import { deepOrange } from "@mui/material/colors"
+
+const Coment: React.FC<{ items: string[] }> = (props) => {
```
Comment (irenlian) [inline src/components/Coments/Coment.tsx:6]:
```suggestion
interface Props {
  items: string[]
}
const Comment: React.FC<Props> = ({ items }) => {
```

File: src/components/Coments/Coment.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    return <div className='wrapperComent'>
+        <div className='comentMessage'>
+            {props.items.map((item, i) => (<div className='comentItem'>
+                <div className='userAvatar'>
+                    <Avatar src='' alt='User avatar' sx={{ width: 20, height: 20, bgcolor: deepOrange[500] }} />
+                    <p className='nickname'>User nickname</p>
+                </div>
+                <TextField multiline key={item[i]} margin='dense' disabled variant='outlined' defaultValue={props.items[i]} ></TextField></div>
+            ))}
```
Comment (irenlian) [inline src/components/Coments/Coment.tsx:15]: During `map` you need to pass a key to the element.
And you should use an item inside, not requesting it once again using an index.

```suggestion
            {props.items.map((item, i) => (
            <div className='comentItem' key={item.id} >
                <div className='userAvatar'>
                    <Avatar src='' alt='User avatar' sx={{ width: 20, height: 20, bgcolor: deepOrange[500] }} />
                    <p className='nickname'>User nickname</p>
                </div>
                <TextField multiline margin='dense' disabled variant='outlined' defaultValue={item} ></TextField>
                </div>
            ))}
```

File: src/components/Coments/Coments.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    flex-direction: column;
+    align-items: center;
+}
+
+.errorMsg {
+    margin: 0px;
+    color: red;
+}
+.hidden{
+visibility: hidden;
```
Comment (irenlian) [inline src/components/Coments/Coments.css:12]:
```suggestion
    visibility: hidden;
```

File: src/components/Coments/Coments.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import Button from '@mui/material/Button'
+import { TextField } from '@material-ui/core'
+import AddCommentTwoToneIcon from '@mui/icons-material/AddCommentTwoTone';
+import './Coments.css'
+import React, { useEffect, useState } from 'react';
+
+const Coments: React.FC<{ message: any }> = (props) => {
```
Comment (irenlian) [inline src/components/Coments/Coments.tsx:7]: Don't use `any`.

File: src/components/Coments/Coments.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    const [newMessage, setNewMessage] = useState('')
+    const [isEmpty, setIsEmpty] = useState(true)
+    const [isToughc, setIsToughc] = useState(false)
+    const showError: boolean = isEmpty && isToughc
+    let isEmptyCopy: boolean = isEmpty
+    let touch: boolean = isToughc
+    let messageCopy: string = newMessage
+
+    function changeHandler(event: any) {
```
Comment (irenlian) [inline src/components/Coments/Coments.tsx:16]: Check the React events types

File: src/components/Coments/Coments.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            setIsEmpty(isEmptyCopy)
+            messageCopy = ''
+            setNewMessage(messageCopy)
+        }
+
+    }
+
+    useEffect(() => {
+
+    }, [touch, isEmptyCopy, messageCopy])
```
Comment (irenlian) [inline src/components/Coments/Coments.tsx:47]: Why do you rely on these variables and not state?

File: src/components/Coments/Coments.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        touch = true
+        setIsToughc(touch)
+        isEmptyCopy = false
+        setIsEmpty(isEmptyCopy)
+
+    }
+
+    function blurHandler() {
+        setIsToughc(true)
+        if (newMessage.trim.length > 0) {
```
Comment (irenlian) [inline src/components/Coments/Coments.tsx:27]:
```suggestion
        if (newMessage.trim().length > 0) {
```

File: src/components/Coments/ComentsTotal.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import Coments from "./Coments";
+import Coment from "./Coment";
+import { useEffect, useState } from "react";
+
+const ComentsTotal: React.FC = () => {
+  const [comentArr, setComentArr] = useState<string[]>(
+    ['Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'])//Fetch с сервера
+
+  const itemList: string[] = [...comentArr]
```
Comment (irenlian) [inline src/components/Coments/ComentsTotal.tsx:11]: The same question - what is it for?

**Review by irenlian — COMMENTED** [review]
Here are some comments.
And I suggest checking if prettier works in your IDE.

**Review by Pashalevchenko — APPROVED** [review]

**Pull request closed without merging**